yivi-core/index.ts

```diff
@@ -1,7 +1,13 @@
-const StateMachine = require('./state-machine');
+import StateMachine  from './state-machine';
 
-module.exports = class YiviCore {
-  constructor(options) {
+export default class YiviCore {
+  _modules: any[];
+  _options: Record<string, any>;
+  _stateMachine: StateMachine;
+  _resolve: any;
+  _reject: any;
+
+  constructor(options: Record<string, any>) {
```

Comment:
Oh cool! I never knew this type existed. 😅 I always use something like `{ [key: string]: any }`, but the `Record` type seems way better :)

(This comment can be resolved. It's just something I found interesting.)